android/app/src/main/java/com/zulipmobile/notifications/NotifyReact.kt

```diff
@@ -56,10 +56,15 @@ private fun emitOrLaunch(application: ReactApplication, eventName: String, data:
         return
     }
 
-    if (reactContext.lifecycleState != LifecycleState.RESUMED) {
-        launchMainActivity(application as Context)
-    } else {
-        emit(reactContext, eventName, data)
+    val lifecycleState = reactContext.lifecycleState!!
```

Comment:
The `!!` seems redundant: is this actually nullable?

Reply:
That's a fun bit of Kotlin-Java interop!
https://kotlinlang.org/docs/reference/java-interop.html#null-safety-and-platform-types

In short: it is, because in Java all object types are implicitly nullable.

The IDE is smart enough that (a) if this were redundant, you'd see a yellow "warning" highlight on the operator, and Alt+Enter would remove it; (b) if on the other hand you do remove it, you get a yellow warning on the `when` because it's no longer exhaustive, which you can resolve by adding a case for `null`.

Reply:
Ah! I did not in fact receive a warning when I deleted the `!!`. Likely because I need to configure something in my IDE.

Reply:
Hmm!

I still get it at the tip of the branch in this version. Would be well worth investigating and sorting out what config that is -- and then getting it into the repo so everyone gets it reliably.